sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -457,7 +477,7 @@ object PartitioningUtils {
     val dateTry = Try {
       // try and parse the date, if no exception occurs this is a candidate to be resolved as
       // DateType
-      DateTimeUtils.getThreadLocalDateFormat(DateTimeUtils.defaultTimeZone()).parse(raw)
+      dateFormatter.parse(raw)
```

Comment:
I'm not sure if the perf matters here, but shall we keep it same and cache the `timestampFormatter` by timezone?

Reply:
I don't think it makes sense to cache `timestampFormatter` because it is created before parsing of all partitions: https://github.com/apache/spark/pull/23391/files#diff-82f61c59249b9d78e6dbf943e19ed043R128